[WIP] JAVA-1069: Bootstrap driver-examples module

adutra opened the pull request.

olim7t committed:
JAVA-843: Disable frozen checks in mapper.

Backported from 3.0 branch.

Commit:
JAVA-444: Add Java process information to UUIDs.makeNode() hash.

Commit:
JAVA-819: Expose more errors in RetryPolicy + provide idempotent-aware wrapper.

Commit:
JAVA-1005: DowngradingConsistencyRetryPolicy does not work with EACH_QUORUM when 1 DC is down.

wy96f committed:
JAVA-1038: Fetch node info by rpc_address if its broadcast_address is not in system.peers.

olim7t committed:
JAVA-833: Improve message when a nested type can't be serialized.

ifesdjeen committed:
Add CREATE KEYSPACE statement builder.

ifesdjeen committed:
Add ALTER KEYSPACE statement builder.

ifesdjeen committed:
Add DROP KEYSPACE statement builder.

olim7t committed:
Update doc links in readme.

Point to manual instead of javadocs.

olim7t committed:
Update doc links in readme.

Point to manual instead of javadocs.

olim7t committed:
Update connection pooling doc to link to LBP.

olim7t committed:
Update connection pooling doc to link to LBP.

Commit:
JAVA-1064: getTable create statement doesn't properly handle quotes in primary key.

Commit:
Minor enhancements to CCM tests.

Highlights:
- Apply annotation-level configuration to CCM builders obtained via ccmProvider.
- Generic way to randomize ports in a CCMBridge.Builder instance.
- Generate CCM cluster names only when building CCMBridge instances.
- Improve algorithm to find available ports.

Commit:
Update clirr-ignores.xml with false positives due to optional compilation of JDK8-only classes.

djatnieks committed:
JAVA-1068: Unwrap StatementWrappers when hashing the paging state.

Commit:
Rename submodule driver-examples to driver-tests.

Commit:
Bootstrap driver-examples module.

Commit:
Configure logging.

Commit:
Add first basic examples.

olim7t committed:
JAVA-888: Add cluster-wide percentile tracker.

olim7t committed:
JAVA-963: Automatically register PercentileTracker from components that use it.

olim7t commented on driver-examples/src/main/java/com/datastax/driver/examples/basic/Example0001.java:
```diff
@@ -0,0 +1,80 @@
+ *
+ * @see <a href="http://datastax.github.io/java-driver/manual/">Java driver online manual</a>
+ */
+public class Example0001 {
```
It would be nice if the class names gave some indication of what they do:
- Example0001 => ReadCassandraVersion
- Example0002 => ReadHostsAndTablesMetadata
- Example0003 => CreateAndPopulateSchema

Numbering the examples by increasing order of complexity is compelling, but it might not scale well: if the counter is at 500 and we realize a simple example is missing, there will be no way to insert it.

olim7t commented on driver-examples/src/main/java/com/datastax/driver/examples/basic/Example0003.java:
```diff
@@ -0,0 +1,143 @@
+/**
+ * This example shows how to create schema objects,
+ * load data into Cassandra then retrieve and display them.
+ *
```
We could have a section here that indicates what each test mutates in the database.

olim7t commented on driver-examples/src/main/java/com/datastax/driver/examples/basic/Example0001.java:
```diff
@@ -0,0 +1,80 @@
+        try {
+
+            cluster = Cluster.builder()                                                    // (1)
+                    .addContactPoint("127.0.0.1")
```
Let's use `addContactPoints(String...)` and extract the array into a `CONTACT_POINTS` constant that will be the same in all examples. And also add a `PORT` constant.

olim7t committed:
Add lifecycle-aware versions of LatencyTracker and Host.StateListener.

Commit:
JAVA-764: Retry with the normal consistency level (not the serial one) when a write times out on the Paxos phase.

Commit:
JAVA-1011: Expose PoolingOptions defaults values.

Commit:
JAVA-727: Allow monotonic timestamp generators to drift in the future + use microsecond precision when possible.

This commit introduces a new dependency to JNR, and uses the system call
'gettimeofday' when available to generate timestamps with microsecond
precision.

olim7t committed:
Clarify javadoc of connectionErrors metric.

olim7t committed:
Clarify javadoc of connectionErrors metric.

Commit:
JAVA-852: Ignore peers with null entries during auto-discovery.

olim7t committed:
JAVA-630: Don't process DOWN events for nodes that have active connections.

Commit:
JAVA-979: Update javadoc for RegularStatement toString() and getQueryString() to indicate that consistency level and other parameters are not maintained in the query string.

Commit:
JAVA-923: Position idempotent flag on object mapper queries.

- make all queries generated by Mapper idempotent;
- add flag to QueryParameters annotation to configure Accessor queries.

Commit:
JAVA-831: Mapper can't load an entity where the PK is a UDT.

Commit:
JAVA-974: Validate accessor parameter types against bound statement.

olim7t committed:
JAVA-851: Improve UUIDs javadoc with regard to user-provided timestamps.

Commit:
JAVA-1070: The Mapper should not prepare queries synchronously.

olim7t committed:
JAVA-1083: Document server-side RPC address configuration.

olim7t committed:
Merge pull request #594 from datastax/java845

JAVA-845: Backport JAVA-843 to 2.1

olim7t committed:
Merge pull request #555 from datastax/java833

JAVA-833: Improve message when a nested type can't be serialized.

olim7t committed:
Merge pull request #589 from datastax/java1011

JAVA-1011: Expose PoolingOptions defaults values.

olim7t committed:
JAVA-1011: Expose PoolingOptions defaults values.

(cherry picked from 2.1)

olim7t committed:
JAVA-630: Don't process DOWN events for nodes that have active connections.

(cherry picked from 2.1)

olim7t committed:
Merge pull request #595 from datastax/java630

JAVA-630: Don't process DOWN events for nodes that have active connec…

olim7t committed:
JAVA-851: Improve UUIDs javadoc with regard to user-provided timestamps.

(cherry picked from 2.1)

olim7t committed:
Merge pull request #602 from datastax/java851

JAVA-851: Improve UUIDs javadoc with regard to user-provided timestamps.

olim7t committed:
Merge pull request #597 from datastax/java979

JAVA-979: Update javadoc for RegularStatement

olim7t committed:
JAVA-979: Update javadoc for RegularStatement

(cherry picked from 2.1)

olim7t committed:
Merge pull request #596 from datastax/java1038

JAVA-1038: Fetch node info by rpc_address if it has a broadcast_address which is not in system.peers.

Commit:
JAVA-879: Mapper.map() accepts mapper-generated and user queries.

olim7t committed:
JAVA-1072: Ensure defunct connections are properly evicted from the pool.

- defunct connection when setKeyspaceAsync call fails.
- make code more robust in case close() is called before defunct().

olim7t committed:
JAVA-1021: Improve error message when connect() is called with an invalid keyspace name.

olim7t committed:
Merge pull request #601 from datastax/java974

JAVA-974: Validate accessor parameter types against bound statement.

tolbertam committed:
Define isolated profile that disables tests at parent level.

The 'isolated' profile was introduced in driver-core a while back to run
tests in their own JVM fork when needed.  This causes problems however
when another submodule like driver-mapping doesn't have an isolated
profile in which it will run unit tests and overwrite any existing
surefire report.  This change disables tests when running with isolated
group unless you override it explicitly in a module that desires to use
it.

tolbertam committed:
Fix should_fail_if_wrong_parameter_type for 1.2.

Use positional parameters instead of named parameters (aren't supported
in Cassandra 1.2).

olim7t committed:
JAVA-1094: Backport TypeCodec parse and format fixes from 3.0.

tommystendahl committed:
JAVA-1040: SimpleStatement parameters support in QueryLogger

tommystendahl committed:
JAVA-1040: SimpleStatement parameters support in QueryLogger.

olim7t committed:
JAVA-1068: Unwrap StatementWrappers when hashing the paging state.

(cherry picked from 2.1)

olim7t committed:
Merge pull request #605 from datastax/java1068

JAVA-1068: Unwrap StatementWrappers when hashing the paging state.

emolsson committed:
JAVA-1100: Exception when connecting with shaded java driver in OSGI

Commit:
JAVA-1089: Set LWT made from BuiltStatements to non-idempotent.

olim7t committed:
Merge pull request #600 from datastax/java831

JAVA-831: Mapper can't load an entity where the PK is a UDT.

olim7t committed:
JAVA-1021: Improve error message when connect() is called with an invalid keyspace name.

(cherry picked from 2.1)

olim7t committed:
Merge pull request #607 from datastax/java1021

JAVA-1021: Improve error message when connect() is called with an inv…

olim7t committed:
JAVA-1083: Document server-side RPC address configuration.

(cherry picked from 2.1)

olim7t committed:
Merge pull request #604 from datastax/java1083

JAVA-1083: Document server-side RPC address configuration.

Commit:
Fix visibility issues.

Members of package-private classes should not be public.

Commit:
Unify reserved keywords list in Metadata and SchemaStatement. Fix SchemaStatement.validateNotKeyword().

Commit:
Make QueryBuilder.quote() delegate to Metadata.quote().

Commit:
Check for null inputs.

Commit:
Fix failing test + small enhancement to javadoc.

Commit:
JAVA-982: Introduce new method ConsistencyLevel.isSerial().

olim7t committed:
Merge pull request #608 from datastax/java879

JAVA-879: Mapper.map() accepts mapper-generated and user queries.

olim7t committed:
JAVA-879: Mapper.map() accepts mapper-generated and user queries.

(cherry picked from 2.1)

olim7t committed:
JAVA-1100: Exception when connecting with shaded java driver in OSGI

(cherry picked from 2.1)

olim7t committed:
Merge branch 'emolsson-JAVA-1100-2.1' into 2.1

olim7t committed:
Merge pull request #587 from datastax/java1050

JAVA-1050: Backport JAVA-819 to 2.1

tolbertam committed:
Add should_log_queries_beyond_dynamic_threshold.

tolbertam committed:
Add tests for PercentileTrackers.

olim7t committed:
Merge branch 'java1040' into 2.1

tolbertam committed:
Add unit tests for UUIDs and Clock.

tolbertam committed:
Mark SimpleStatement w/ Parameters tests as requiring C* 2.0

olim7t committed:
JAVA-1064: getTable create statement doesn't properly handle quotes in primary key.

(cherry picked from 2.1)

olim7t committed:
Merge pull request #582 from datastax/java1064

JAVA-1064: getTable create statement doesn't properly handle quotes in primary key.

olim7t committed:
Merge pull request #586 from datastax/java1051

Percentile tracker improvements

olim7t committed:
Merge pull request #613 from datastax/java1089

JAVA-1089: Set LWT made from BuiltStatements to non-idempotent.

olim7t committed:
JAVA-1089: Set LWT made from BuiltStatements to non-idempotent.

(cherry picked from 2.1)

olim7t committed:
JAVA-923: Position idempotent flag on object mapper queries.

(cherry picked from 2.1)

olim7t committed:
Merge pull request #599 from datastax/java923

JAVA-923: Position idempotent flag on object mapper queries.

tolbertam committed:
Upgrade to scassandra 1.0.3.  Change CAS tests to use Scassandra.

olim7t committed:
Merge branch 'java1019' into 2.1

olim7t committed:
JAVA-1070: The Mapper should not prepare queries synchronously.

(cherry picked from 2.1)

olim7t committed:
Merge pull request #603 from datastax/java1070

JAVA-1070: The Mapper should not prepare queries synchronously.

olim7t committed:
Define isolated profile that disables tests at parent level.

(cherry picked from 2.1)

Also cherry-picked
MapperAccessorParamsTest.should_fail_if_wrong_parameter_type
coming from JAVA-974 (974 didn't need to be merged to 3.0.x because it
was already covered with the new codec mechanism, but it doesn't hurt to
have the test).

olim7t committed:
Merge pull request #609 from datastax/fix-mapper-reporting

Define isolated profile that disables tests at parent level.

olim7t committed:
Merge pull request #588 from datastax/java764

JAVA-764: LWT / CAS Consistency bug on Retry Policy.

olim7t committed:
JAVA-764: LWT / CAS Consistency bug on Retry Policy.

(cherry picked from 2.1)

olim7t committed:
Fix logger category for Mapper.

(cherry picked from 2.1)

olim7t committed:
Fix logger category for Mapper.

tolbertam committed:
Fix issue where tests were being skipped in CI.

Because we have @Before* methods on CCMTestsSupport, any test class
extending this class that had a @CassandraVersion annotation which
causes tests to be skipped for that class would also cause subsequent
tests to be skipped, even if they lacked the @CassandraVersion
annotation.

This happens because TestNG takes the declaring Class of the method
being skipped instead of the Class implementation being evaluated.  To
bypass this, setting configfailurepolicy to true causes TestNG to
evaluate the class implementation instead of the declaring class of the
@Before-annotated method.

Also simplifies build by declaring profiles in parent pom,
and use test.groups property for declaring test groups.

Version mapper tests based on features used. (UDTs, Named params)

tolbertam committed:
Tighten log validation in QueryLoggerTest.

Was checking on numeric values that could appear in timestamps and port
numbers.  Tighten this to prevent false failures.

Commit:
Declare failsafe plugin at root level. Bump plugin version to latest.

Also fix some failing tests.

olim7t committed:
JAVA-977: Preserve original cause when BuiltStatement value can't be serialized.

olim7t committed:
Merge pull request #591 from datastax/java727

JAVA-727: Allow monotonic timestamp generators to drift in the future + use microsecond precision when possible.

olim7t committed:
Merge pull request #593 from datastax/java444

JAVA-444: Add Java process information to UUIDs.makeNode() hash.

olim7t committed:
Add test to cover JAVA-977.

olim7t committed:
Merge pull request #619 from datastax/java977

JAVA-977: Preserve original cause when BuiltStatement value can't be serialized.

olim7t committed:
Merge pull request #618 from datastax/java1094

JAVA-1094: Backport TypeCodec parse and format fixes from 3.0.

tolbertam committed:
Add tests for JAVA-852.

olim7t committed:
JAVA-852: Ignore peers with null entries during auto-discovery.

(cherry picked from 2.1)

olim7t committed:
Merge pull request #592 from datastax/java852

JAVA-852: Ignore peers with null entries during auto-discovery.

tolbertam committed:
Fix schema agreement bug in JAVA-852.

The isValidPeer check in checkSchemaAgreement requires the presence of
columns that weren't returned in SELECT_SCHEMA_PEERS.  Updated to use
SELECT_PEERS as we want to only valid peers are considered.

tolbertam committed:
Separate out EXTENDED_PEER_CHECK tests so they may run in isolated mode.

tolbertam committed:
Include schema_version in primed peers and local queries.

olim7t committed:
Merge pull request #621 from datastax/java852-fix

Fix checkSchemaAgreement bug introduced in JAVA-852

olim7t committed:
Move idempotence to its own page.

olim7t committed:
Move idempotence doc to its own page.

(cherry picked from 2.1)

tolbertam committed:
Merge pull request #585 from datastax/ccm-minor-enhancements

More enhancements to CCM tests.

tolbertam committed:
More enhancements to CCM tests.

(cherry picked from 2.1)

tolbertam committed:
Merge pull request #620 from datastax/java764tests

Upgrade to scassandra 1.0.3.  Change CAS tests to use Scassandra.

tolbertam committed:
Upgrade to scassandra 1.0.3.  Change CAS tests to use Scassandra.

(cherry picked from 2.1)

olim7t committed:
Document retries.

(cherry picked from 2.1)

olim7t committed:
Document retries.

olim7t committed:
Backport javadoc config changes from 3.0.

olim7t committed:
Document PercentileTracker changes.

olim7t committed:
Link to retries in specex doc.

olim7t committed:
Link to retries in specex doc.

olim7t committed:
Document that JNR doesn't work with OSGI.

olim7t committed:
Remove obsolete links.

(cherry picked from 2.1)

olim7t committed:
Remove obsolete links.

tolbertam committed:
Allow multiple workloads to specified for a node.

tolbertam committed:
Add updateDSENodeConfig as updatedseconf now supports node level ops.

Commit:
JAVA-1132: Executing bound statement with no variables results in exception with protocol v1.

Commit:
JAVA-1132: Executing bound statement with no variables results in exception with protocol v1.

olim7t committed:
Expose ConnectionException and subclasses.

olim7t committed:
Merge pull request #624 from datastax/2.1.10_doc

2.1.10 doc

Commit:
Allow multiple workloads to be specified per node in @CCMConfig.

olim7t committed:
Merge pull request #627 from datastax/java1132-2.1

JAVA-1132 [2.1]: Executing bound statement with no variables results in exception with protocol v1.

olim7t committed:
JAVA-1005: DowngradingConsistencyRetryPolicy does not work with EACH_QUORUM when 1 DC is down.

(cherry picked from 2.1)

olim7t committed:
Merge pull request #544 from datastax/java1005

JAVA-1005: DowngradingConsistencyRetryPolicy does not work with EACH_QUORUM when 1 DC is down.

olim7t committed:
JAVA-1002: Avoid deadlock when re-preparing a statement on other hosts.

olim7t committed:
Merge pull request #629 from datastax/java1002

JAVA-1002: Avoid deadlock when re-preparing a statement on other hosts.

olim7t committed:
JAVA-1002: Avoid deadlock when re-preparing a statement on other hosts.

(cherry picked from 2.1)

olim7t committed:
Prepare changelog for 3.0.1.

olim7t committed:
Merge pull request #606 from datastax/java1072

JAVA-1072: Defunct connection when setKeyspaceAsync call fails.

olim7t committed:
Merge branch '2.0' into 2.1

olim7t committed:
JAVA-1072: Ensure defunct connections are properly evicted from the pool.

(cherry picked from 2.1)

Commit:
Retro-fitted FAQs from DITA to Documentor ...

olim7t committed:
Merge pull request #640 from datastax/docs-2.1

Retro-fitted FAQs from DITA to Documentor ...

olim7t committed:
Retro-fitted FAQs from DITA to Documentor ...

(cherry picked from 2.1)

olim7t committed:
Update doc for 2.1.10 release.

olim7t committed:
[maven-release-plugin] prepare release 2.1.10

olim7t committed:
[maven-release-plugin] prepare for next development iteration

olim7t committed:
Force the underlying connection in ControlConnection.closeAsync().

Otherwise the connection's future might not complete properly, if there
was an in-flight request at the time when closeAsync() was called.

Also force() in other places where we close the connection, to ensure
the channel gets closed.

olim7t committed:
Prepare merge from 2.1.

Empty commit as the changes will be cherry-picked manually.

olim7t committed:
Format POMs with Intellij defaults.

olim7t committed:
Merge pull request #634 from datastax/java1132-3.0

JAVA-1132 [3.0]: Executing bound statement with no variables results in exception with protocol v1.

olim7t committed:
Merge pull request #625 from datastax/multi-workload-support

Support multiple workloads for DSE nodes, updateDSENodeConfig

olim7t committed:
Merge branch '2.1' into 3.0.x

olim7t committed:
Document where reserved keyword list comes from.

olim7t committed:
Fix typo in mapper doc.

olim7t committed:
Merge branch 'tommystendahl-java1040_3.0' into 3.0.x

olim7t committed:
Address my feedback.

olim7t committed:
JAVA-1150: Add example and FAQ entry about ByteBuffer/BLOB.

olim7t commented:
Closing as this will now target 3.0.x.
